lib/pyfrc/mains/cli_deploy.py

```diff
@@ -126,11 +127,12 @@ def run(self, options, robot_class, **static_options):
             bash_cmd = '/bin/bash -ce'
 
         if options.in_place:
-            del_cmd = ''
+            replace_cmd = 'true'
+            py_new_deploy_dir = py_deploy_dir
         else:
-            del_cmd = "[ -d %(py_deploy_dir)s ] && rm -rf %(py_deploy_dir)s"
+            replace_cmd = "[ -d %(py_deploy_dir)s ] && rm -rf %(py_deploy_dir)s && mv %(py_new_deploy_dir)s %(py_deploy_dir)s"
```

Comment:
the remove command is recursive, touches lots of files. if it fails, the new code won't be moved to the code directory, and the old code will be in a possibly invalid state.
can't tell from my quick read of this file, will this command failing result in the entire deploy failing?

Reply:
Yes. I'd be pretty hard-pressed to find a scenario where the `rm -rf` would fail.

I just realised though that the new code wouldn't be moved to its desired location like this though if no code was deployed. I'll, uh, fix that now.